### PR TITLE
Update language-specific docs to document that public functions can be called from the backend code

### PR DESCRIPTION
**Pull request by danutsu**
Tested for Rust and C++, wasn't able to test JS but it should work.

I found a `sample_component.slint` file that seems to be intended to be the same code as the SampleComponent in Rust and C++ docs, so I updated it to be in sync with the code in those language docs. Let me know if that is not what it's supposed to be.

### CODE REVIEW

File: api/rs/slint/docs.rs [hunk 1]
```diff
@@ -23,13 +23,16 @@ pub mod generated_code {
 
     /// This an example of the API that is generated for a component in `.slint` design markup. This may help you understand
     /// what functions you can call and how you can pass data in and out.
+    /// 
     /// This is the source code:
+    ///
```
Comment (tronical) [inline api/rs/slint/docs.rs:28]:
```suggestion
    ///
```

File: api/cpp/docs/generated_code.md [hunk 2]
```diff
@@ -26,6 +26,8 @@ These classes have the same name as the component will have the following public
   * `invoke_<callback_name>` function which takes the callback argument as parameter and call the callback.
   * `on_<callback_name>` function which takes a functor as an argument and sets the callback handler
      for this callback. the functor must accept the type parameter of the callback
+* For each public function declared in the root component, an `invoke_<function_name>` function which
+  takes the function arguments as parameters and calls the function.
```
Comment (tronical) [inline api/cpp/docs/generated_code.md:30]: Suggestion for simplification. I think it's fair of users to expect that argument are passed along.

```suggestion
* For each public function declared in the root component, an `invoke_<function_name>` function to call the function.
```

**Review by tronical — APPROVED** [review]
Looks overall good to me. Thanks! What do you think of the supplied suggestion?

**Comment by danutsu** [comment]
Sure :) I was just following the pattern above but I agree it's pretty redundant. And thanks for the other fixes too, I should have been more careful.

Let me see if CI passes now, I think the first time it failed with some sort of CI infrastructure issue.

**Pull request merged**